version: "3.5"
   
services:
  database:
   image: kartoza/postgis:9.6-2.4
   volumes:
      - ./data/db:/var/lib/postgresql/data
   environment:
      - POSTGRES_USER=gisuser
      - POSTGRES_DB=gis
      - POSTGRES_PASS=/run/secrets/message
   ports:
   - "5432:5432"
   secrets:
      - message


  backend:
     build: 
      context: backend
      dockerfile: Dockerfile
     container_name: backend
     command: bash -c "python manage.py migrate && python manage.py runserver"
     environment:
       DATABASE_HOST: database
       POSTGRES_USER: gisuser
       POSTGRES_DB: gisuser
       AWS_ACCES_KEY_ID: $$AWS_ACCES_KEY_ID
       AWS_SECRET_ACCES_KEY: $$AWS_SECRET_ACCES_KEY
       ENVIRONMENT: $$ENVIRONMENT
       MAIL_USER: $$MAIL_USER
       MAIL_KEY: $$MAIL_KEY
       EMAIL_REGISTER: $$EMAIL_REGISTER
       SECRET_KEY: $$SECRET_KEY
       POSTGRES_PASS: /run/secrets/message

     volumes:
       - ./backend/app/:/src/app
     ports:
     - "8000:8000"
     restart: on-failure
     links:
       - database
     secrets:
      - message
secrets:
  message:
    name: "arn:aws:ssm:sa-east-1:612487716168:parameter/POSTGRES_PASS"
    external: true